portfolio/src/main/webapp/galleryScript.js

```diff
@@ -0,0 +1,38 @@
+function openGallery() {
+	document.getElementById('myGallery').style.display = 'block';
+}
+
+function closeGallery() {
+	document.getElementById('myGallery').style.display = 'none';
+}
+
+var slideIndex = 1;
+showSlidingGallery(slideIndex);
+
+function moveNSlides(n) {
+	showSlidingGallery(slideIndex += n);
+}
+
+function showCurrentSlide(n) {
+	showSlidingGallery(slideIndex = n);
+}
+
+function showSlidingGallery(n) {
```

Comment:
So this function is getting a lot of hard-coded data out of the HTML - like the description text, alt text, and so on.

Something that might be worth thinking about is a way of storing the same data in an array variable, for example:

```
const images = [
  {
    description: 'description 1',
    alt: 'alt 1',
    src: 'images/1.jpg',
  },
  {
    description: 'description 2',
    alt: 'alt 2',
    src: 'images/2.jpg',
  },
];
```
So here you have an array called `images`, and each element of the array contains an object with all the bits you need for your images that you can just access using `.src`, `.description` etc, e.g. 

```
let currentImage = images[1]
console.log(currentImage.src); // will print 'images/2.jpg'
```

So you can iterate over this array instead of pulling things out of the document. The real benefit is that you can *change* what is in the `images` array at run-time ... so in this code at the moment it is just hard-coded, but if you had a list of images on the sever you could fetch that info and then update `images` based on what comes back from the server.

Anyway - just something to think about that might come in handy over the next couple of weeks ... :-)  Give me a shout if you'd like me to explain anything more.

Reply:
Oh yes that makes a lot of sense, I didn't think about that